CesiumGeometry/test/TestIntersectionTests.cpp

```diff
@@ -35,3 +35,214 @@ TEST_CASE("IntersectionTests::rayPlane") {
+      TestCase{
+          glm::dvec3(-2.0, 0.5, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          false});
+
+  bool result = IntersectionTests::pointInTriangle(
+      testCase.point,
+      testCase.triangleVert1,
+      testCase.triangleVert2,
+      testCase.triangleVert3);
+  CHECK(result == testCase.expected);
+}
+
+TEST_CASE("IntersectionTests::pointInTriangle with barycentric coordinates") {
+  struct TestCase {
+    glm::dvec3 point;
+    glm::dvec3 triangleVert1;
+    glm::dvec3 triangleVert2;
+    glm::dvec3 triangleVert3;
+    bool expected;
+    glm::dvec3 expectedCoordinates;
+  };
+
+  auto testCase = GENERATE(
+      // is in triangle (corner)
+      TestCase{
+          glm::dvec3(1.0, 0.0, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          true,
+          glm::dvec3(0.0, 0.0, 1.0)},
+      // is in triangle (edge)
+      TestCase{
+          glm::dvec3(0.0, 0.0, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          true,
+          glm::dvec3(0.5, 0.0, 0.5)},
+      // is in triangle (middle)
+      TestCase{
+          glm::dvec3(0.0, 0.5, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          true,
+          glm::dvec3(0.25, 0.5, 0.25)},
+      // is in triangle (regardless of winding order)
+      TestCase{
+          glm::dvec3(0.0, 0.5, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          true,
+          glm::dvec3(0.25, 0.25, 0.5)},
+      // is not in triangle (same plane)
+      TestCase{
+          glm::dvec3(-2.0, 0.5, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          false,
+          glm::dvec3()},
+      // is not in triangle (different plane)
+      TestCase{
+          glm::dvec3(0.2, 0.5, 1.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          false,
+          glm::dvec3()},
+      // is not in triangle (regardless of winding order)
+      TestCase{
+          glm::dvec3(-2.0, 0.5, 0.0),
+          glm::dvec3(-1.0, 0.0, 0.0),
+          glm::dvec3(1.0, 0.0, 0.0),
+          glm::dvec3(0.0, 1.0, 0.0),
+          false,
+          glm::dvec3()});
+
+  glm::dvec3 barycentricCoordinates;
+  bool result = IntersectionTests::pointInTriangle(
+      testCase.point,
+      testCase.triangleVert1,
+      testCase.triangleVert2,
+      testCase.triangleVert3,
+      barycentricCoordinates);
+
+  REQUIRE(result == testCase.expected);
+  if (result) {
+    CHECK(barycentricCoordinates == testCase.expectedCoordinates);
```

Comment:
We should probably always check that the returned `barycentricCoordinates` is what we expect, even when the function returns `false`.